### PR TITLE
fix(api): stop using env cookie

### PR DESCRIPTION
**Pull request by bodinsamuel**
## Describe your changes

Fixes NAN-729

We were using cookie as a way to automically detect env, because cookies are shared across tab this could create issues when switching env in an other tab and doing a request in an other.

- Remove use of cookie env, all (almost) path now take `env` as a query parameter

Tested on staging but please test on your side since it's touching every pages and endpoints (edit, deleted and stuff)

### CODE REVIEW

File: packages/server/lib/utils/utils.ts [hunk 1]
```diff
@@ -1,41 +1,52 @@
...
 
-    const environmentAndAccount = await environmentService.getAccountAndEnvironmentById(user.account_id, currentEnvironment);
+    return resultOk(user);
+}
 
+export async function getUserAccountAndEnvironmentFromSession(
+    req: Request<any, any>
+): Promise<ServiceResponse<{ user: User; account: Account; environment: Environment }>> {
+    const getUser = await getUserFromSession(req);
+    if (!getUser.ok) {
```
Comment (TBonnin) [inline packages/server/lib/utils/utils.ts]: maybe we should agree on an unique way to check for failure. For example always using `isErr` or `isOk`. I think the consistency will make code easier to read on the long term

**Review by khaliqgant — CHANGES_REQUESTED** [review]
Trying to create a connection results in 
image

**Comment by bodinsamuel** [comment]
> Looks like the frontend sdk calls need to set an env as well.

There is a deeper issue here, because it shouldn't be the case. There is some shared code that shouldn't shared across different auth strategy

**Comment by bodinsamuel** [comment]
Can I get a review again? @khaliqgant @TBonnin

**Review by TBonnin — APPROVED** [review]

File: packages/server/lib/utils/utils.ts (hunk 1 above)
Comment (bodinsamuel) [reply packages/server/lib/utils/utils.ts]: ah yeah, it just works like this so I don't always think about the helpers

**Review by khaliqgant — APPROVED** [review]
Code looks good. I haven’t had a chance to pull again locally to test but feel free to merge if you want. If not I’ll test again this evening

File: packages/server/lib/utils/utils.ts (hunk 1 above)
Comment (bodinsamuel) [reply packages/server/lib/utils/utils.ts]: Fixed*

**Pull request merged**